### PR TITLE
LZ-SA-estimated-next-purchase

### LINKED ISSUE #10: 10. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database
AC:

* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

The following script can be used to calculate the estimated next purchase date: https://gist.github.com/segdeha/a45ef8d6d7f094f4038a2318a9d60051

### PR DESCRIPTION
**Pull request by lilyhoratio**
## Description

- In List.js, we calculated the time interval between the item's purchased date (current date) and the `lastPurchaseDate`, and converted that to days. Then, we use the `calculateEstimate` function to recalculate a nextPurchase interval for that item based on a weighted average using the prior nextPurchase interval, actual interval, and number of purchases. When the item is checked (bought), the estimated days interval is used to update the item's `nextPurchase` in firebase. 
- We created a helper function to convert seconds to days in `lib` folder.

## Related Issue

Closes #10 

## Acceptance Criteria

- When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Updates

No UI changes.

## Testing Steps / QA Criteria

1. In the web app, clear out any existing userToken in local storage. Then join an existing list using the token `walton twill litton`
2. In firebase console, filter the shoppingList using condition token == "walton twill litton" to see the items.
3. In firebase console, update the `lastPurchasedDate` of one of the items for that list to be at least 1 day prior to the current date.
4. In the web app, check that item as bought. 
5. Verify that the next purchase date of that item is updated to a newly calculated value.

### CODE REVIEW

File: src/components/List.js [hunk 1]
```diff
@@ -15,15 +17,26 @@ const List = ({ firestore }) => {
   };
 
   const handleChange = (e, item) => {
+    const daysInterval = secondsToDays(
```
Comment (AliLynne) [inline src/components/List.js:20]: Great job naming functions based on their purpose but maybe some comments in this section would make it clearer exactly what's happening.

File: src/lib/secondsToDays.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const secondsToDays = (seconds)  => {
+  const days = Math.floor(Number(seconds) / (3600*24));
```
Comment (segdeha) [inline src/lib/secondsToDays.js:2]: Something to consider: by flooring the value here you lose some fidelity in the calculations. An alternative would be to keep the raw value in the database and floor or round only when you display the value to the user.

File: src/components/List.js [hunk 3]
```diff
@@ -15,15 +17,26 @@ const List = ({ firestore }) => {
...
+      Date.now() / 1000 - item.lastPurchasedDate['seconds'],
+    );
+    const weightedNextPurchaseEstimate = calculateEstimate(
```
Comment (segdeha) [inline src/components/List.js:23]: I'll second what @AliLynne said, great job naming your variables to make it as clear as possible what they represent! 👏👏👏

File: src/components/List.js [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { useHistory } from 'react-router-dom';
 import { FirestoreCollection, withFirestore } from 'react-firestore';
 import Item from './Item';
 import '../styles/List.css';
+import calculateEstimate from '../lib/estimates';
+import secondsToDays from '../lib/secondsToDays';
```
Comment (Lukabaramishvili) [inline src/components/List.js:7]: I love how the `estimates` and `secondsToDays` are abstracted into the lib folder.

**Pull request merged**